modules/cnf-reducing-netqueues-using-pao.adoc

```diff
@@ -0,0 +1,346 @@
+
+In real-time or low latency systems all the unnecessary interrupt request lines (IRQs) that are pinned to the isolated CPUs must be moved to reserved or housekeeping CPUs.
+
+In deployments with applications that require system, {product-title} networking or in mixed deployments with Data Plane Development Kit (DPDK) workloads, multiple queues are needed to achieve good throughput and the number of NIC queues should either be adapted or remain unchanged. For example, to achieve low latency the number of NIC queues for DPDK based workloads should be reduced to just the number of reserved or housekeeping CPUs.
+
+Too many queues are created by default for each CPU and these do not fit into the interrupt tables for house keeping CPUs when tuning for low latency. Reducing the number of queues makes proper tuning possible. Smaller number of queues means smaller number of interrups which then fit in the IRQ table.
+
+[id="adjusting-nic-queues-with-the-performance-profile_{context}"]
+== Adjusting the NIC queues with the performance profile
+
+The performance profile lets you adjust the queue count for each network device.
+
+Supported network devices are:
+
+* Non virtual network devices
+
+* Network devices that support multiple queues (channels)
+
+Unsupported network devices are:
+
+* Pure software network interfaces
+
+* Block devices
+
+* Intel DPDK virtual functions
+
+.Prerequisites
+
+* Access to the cluster as a user with the `cluster-admin` role.
+* Install the OpenShift CLI (`oc`).
+
+
+.Procedure
+
+. Log in to the OpenShift Container Platform cluster running the Performance Addon Operator as a user with `cluster-admin` privileges.
+
+. Create a performance profile that is appropriate for your hardware and topology. For more information see the Tuning nodes for low latency with the performance profile section. Alternatively, edit an existing performance profile using the following command:
++
+[source,terminal]
+----
+$ oc edit performanceprofile <your_profile_name>
+----
+
+. Populate the `spec` field profile with the `net` object. The object list can contain two fields:
+
+* `userLevelNetworking` is a required field that specifies a boolean flag which, if `true` sets the queue count to the reserved CPU count for all supported devices. The default is `false`.
+* `Devices` is an optional field which specifies a list of devices that will have the queues set to the reserved CPU count. If the device list is empty the configuration applies to all network devices. The configuration is as follows:
+** `interfaceName`: Name supports shell-style wildcards which can be positive or negative.
+*** Example wildcard syntax is as follows: `<string> .*`
+*** Negative rules are prefixed with an exclamation mark. To apply the net queue changes to all devices other than the excluded list, use  `!<device>`; for example, `!eno1`.
+* `vendorID`: Network device vendor ID represented as a 16 bit hexmadecimal number with a 0x prefix.
```

Comment:
Looking at the YAML, I think that vendorID and deviceID are supposed to be indented under the bullet for Devices, but they're each a top-level bullet.